dips/dip-10.md

```diff
@@ -13,47 +13,57 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
+* Alice's Diem ID is registered at VASP A: `alice@avasp`
+* Bob shares the Diem ID, `bob@bvasp` with Alice
+* Alice logs into VASP A, enters Bob’s identifier, an amount to pay, and submits payment
+* Alice’s VASP (VASP A) contacts Bob’s VASP’s (VASP B) off-chain API with the sender identifier `alice@avasp` and requests a reference_id, `rb1`
+* VASP A constructs a transaction with the specified amount and the reference_id `rb` and submits it to the Diem network
+* VASP B receives a transaction with metadata containing `rb`, deposits the amount in Bob's account, and attaches the relevant metadata (e.g., Alice’s Diem ID) to the internally stored transaction so Bob can confirm the transaction
+
 
-# Diem ID format
-Diem ID is a string in format `<user-id>$<diem-id-domain>`.
+# Diem ID Format
+Diem defines globally unique identifiers to facilitate transacting on the Diem Payment Network. As Diem stores no PII on-chain, these identifiers are exchanged during the off-chain process between two VASPs to identify a distinct source and destinations within their respective VASPs in order to produce a reference_id that can be stored on-chain. The format for the identifiers follows:
+`[user_identifier]@[vasp_domain_identifier]`
 
-Example: `andrey$novi`
+Example: `alice@avasp`
 
-* **user-id**, identifies an account within a VASP:
+* `user_identifier` is a reusable identifier that represents either the source or destination end-user of a payment transaction. It is unique to per user at VASP level. Specification:
   * Case insensitive
-  * Regex: `^[a-zA-Z0-9][a-zA-Z0-9-_+.]*$`
+  * Valid regular expression: [a-zA-Z0-9]+
   * Maximum length: 64 characters
-  * Stored internally in the VASP. Not published on the chain, there is no way for an external entity to list all the users within VASP.
-  * Defined by the VASP - user selected, auto-generated, etc.
-* **diem-id-domain** identifies a VASP:
+* `vasp_domain_identifier` is a unique string that is mapped to a VASP. Specification:
   * Case insensitive
-  * Regex: `^[a-zA-Z0-9][a-zA-Z0-9-]+$`
+  * Valid regular expression: [a-zA-Z0-9]+
   * Maximum length: 63 characters
-  * Association between diem id and VASP is published on the chain 
 
-## Diem ID domains
-Each Diem ID domain is associated with a single VASP. This information is stored on the block chain in a `DiemIdDomains` resource published in a parent VASP's on-chain account.
+
+# Naming Service (Off-Chain Lookup Service)
+The intent is to eventually store Diem ID domains on-chain; however, the Diem Association will initially store and distribute a CSV file mapping domain name to account address. The account address should be a parent VASP. The VASP sending a payment can then obtain the DualAttestation::Credentials::base_url to perform the ReferenceID exchange. An example CSV is:
+```
+avasp, 0xf72589b71ff4f8d139674a3f7369c69b
+bvasp, 0xc5ab123458df0003415689adbb47326d
+```
 
 # On-chain data
+## Diem ID Domain
```

Comment:
This part of the spec seems to omit defining an easy way to go from Domain -> Account Address. Looking at the "Events" section below also doesn't seem to address this fully since you'd have to look at the stream from the beginning of time in order to reconstruct the up-to-date mapping of Domain -> Account Address.

Reply:
not sure what you mean by the easy way to go from id to account address?

Reply:
I don't think there technically is an easy way, but there are two technical approaches:
* iterate across all accounts and an identify those that have this
* iterate across all events and identify updates or accounts along the way